modules/p2/ding_list/include/functions.inc

```diff
@@ -818,6 +818,11 @@ function ding_list_get_permission_permissions($permission = FALSE) {
 function ding_list_user_has_access($list, $permission = DING_LIST_PERMISSION_VIEW, $account = NULL, $token = FALSE) {
   $access = &drupal_static(__FUNCTION__, array());
 
+  // If used as menu access callback the list have to be loaded.
+  if (is_numeric($list)) {
+    $list = ding_list_get_list($list);
+  }
+
```

Comment:
If we are to utilize the static cache then I would prefer to see this moved below the subsequent access check.

Reply:
Nice catch. Was a very fast PR. :-) Fixed now